components/twitch-panel/twitch-panel.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useMemo, useState } from "react";
+import { Box, Container, Flex, Group, Text, Title, useMantineColorScheme } from "@mantine/core";
+import { IconBrandTwitch, IconCircle } from "@tabler/icons";
+import { TwitchStream } from "../../src/coh3/coh3-types";
+import ChannelList from "./channel-list";
+
+declare global {
+  interface Window {
+    Twitch: any;
+  }
+}
+
+type Props = {
+  twitchStreams: TwitchStream[] | null;
+  error: Error | null;
+};
+const TwitchPanel = ({ twitchStreams, error }: Props) => {
+  const { colorScheme } = useMantineColorScheme();
+  const [player, setPlayer] = useState<any>();
+  const [currentChannelIndex, setCurrentChannelIndex] = useState(0);
+
+  const currentStream = useMemo(() => {
+    return twitchStreams && twitchStreams[currentChannelIndex];
+  }, [twitchStreams, currentChannelIndex]);
+
+  useEffect(() => {
+    // this gate only be needed because of react strict mode running things twice
+    if (document.getElementById("twitch-script") !== null || twitchStreams === null) return;
+    console.log(twitchStreams);
+
+    const script = document.createElement("script");
+    script.src = "https://player.twitch.tv/js/embed/v1.js";
+    script.id = "twitch-script";
+    document.body.appendChild(script);
+
+    script.addEventListener("load", () => {
+      const embed = new window.Twitch.Embed("twitch-embed", {
+        width: "100%",
+        height: "100%",
+        channel: twitchStreams[0].user_login,
+        layout: "video",
+        autoplay: false,
+        theme: colorScheme,
+        // Only needed if this page is going to be embedded on other websites
+        parent: ["embed.example.com", "othersite.example.com"],
+      });
+
+      embed.addEventListener(window.Twitch.Embed.VIDEO_READY, () => {
+        const player = embed.getPlayer();
+        player.play();
+        setPlayer(player);
+      });
+    });
+  }, [twitchStreams, colorScheme]);
+
+  function handleChangeChannel(channelIndex: number) {
+    if (!twitchStreams) return;
+    setCurrentChannelIndex(channelIndex);
+    player.setChannel(twitchStreams[channelIndex].user_login);
+    player.play();
+  }
+
+  return (
+    <Container size="fluid">
+      <Flex justify="flex-start" align="center" gap={10} pb="sm">
+        <IconBrandTwitch size={40} />
+        <Title order={2} size="h2">
+          Watch Live Streams
+        </Title>
+      </Flex>
+
+      <Group>
+        <Container w="80%" h="auto" style={{ aspectRatio: 16 / 8.657 }} maw={854}>
```

Comment:
I think we should not use percentage here, it can look different on different screens. 

If we want some ration let's say 80%/ 20% split some-where we should use something like grid, where you can split it. 

For example I think the 80% is unnecessary on the mobile phone, there we want 100% 
<img width="464" alt="image" src="https://user-images.githubusercontent.com/8086995/226739363-e2448363-3db9-4853-874d-7e84cad6e5d1.png">

Btw it looks pretty good 👍

Reply:
Okay thanks. I am not very good at responsive styling. So its pretty slow going for me. 

My plan is first trying to get it too look decent on desktop first and then try I was going to use the `sx` prop where needed to apply some styles at certain breakpoints.